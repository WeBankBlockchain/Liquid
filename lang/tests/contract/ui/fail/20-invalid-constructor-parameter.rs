use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self, value: f32) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
