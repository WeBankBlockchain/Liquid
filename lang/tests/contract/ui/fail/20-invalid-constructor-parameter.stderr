error[E0277]: the trait bound `f32: You_Should_Use_An_Valid_Input_Type` is not satisfied
  --> $DIR/20-invalid-constructor-parameter.rs:10:38
   |
10 |         pub fn new(&mut self, value: f32) {}
   |                                      ^^^ the trait `You_Should_Use_An_Valid_Input_Type` is not implemented for `f32`
